### PR TITLE
result: add current time for each test. Resolves #104

### LINKED ISSUE #104: Include test failure time in Fail message
Should be an easy enhancement -- append str(datetime.now().time()) to the end of the failure details? This change would help debugging by making it possible to correlate the test failure with the any log messages from the system under test.

### PR DESCRIPTION
**Pull request by andrewbonney**

### CODE REVIEW

File: TestResult.py [hunk 1]
```diff
@@ -24,39 +26,50 @@ def __init__(self, description, name=None):
             self.name = inspect.stack()[1][3]
         self.timer = time.time()
 
+    def _current_time(self):
+        return str(datetime.datetime.now().time())
```
Comment (garethsb-sony) [inline TestResult.py]: I think microseconds is a little excessive (and on my platform the least significant digits are always 333 oddly!)
How about this?
```python
        return datetime.datetime.now().time().isoformat('milliseconds')
```

Comment (andrewbonney) [reply TestResult.py]: For some reason that complained about isoformat not allowing arguments, which is odd given what the docs say. Used an alternative way around it.

**Comment by andrewbonney** [comment]
This should take account of comments from @garethsb-sony and @billt-hlit

File: TestResult.py [hunk 2]
```diff
@@ -27,7 +27,7 @@ def __init__(self, description, name=None):
         self.timer = time.time()
 
     def _current_time(self):
-        return str(datetime.datetime.now().time())
+        return str(datetime.datetime.now().time())[:-3]
```
Comment (garethsb-sony) [inline TestResult.py]: I suggest not doing that.

str(datetime.datetime.now().replace(microsecond=0).time()) == '09:35:14'

Comment (andrewbonney) [reply TestResult.py]: I'm not sure second precision is really enough for this

File: TestResult.py (hunk 1 above)
Comment (garethsb-sony) [reply TestResult.py]: Doesn't work in python 2.7, but works for me with python 3.7 and just tried online at https://www.tutorialspoint.com/execute_python3_online.php which is 3.6

File: TestResult.py (hunk 2 above)
Comment (garethsb-sony) [reply TestResult.py]: Exactly, that's my point

File: TestResult.py (hunk 1 above)
Comment (andrewbonney) [reply TestResult.py]: Ok, well I'm running Python 3.5 here (default for Ubuntu Xenial) which caused an exception.

File: TestResult.py (hunk 2 above)
Comment (andrewbonney) [reply TestResult.py]: Ok, I thought the docs said it would be zero-padded, but if that's not always the case I'll do it another way.

Comment (garethsb-sony) [reply TestResult.py]: [Return a string representing the time in ISO 8601 format, HH:MM:SS.mmmmmm or, if self.microsecond is 0, HH:MM:SS](https://docs.python.org/2/library/datetime.html#datetime.time.isoformat)

Comment (andrewbonney) [reply TestResult.py]: What does the following return on Windows? I'd hope that being explicit with the %f would return six zeros if it has nothing else to put there:

```
datetime.datetime.now().strftime("%H:%M:%S.%f")[:-3]
```

Comment (garethsb-sony) [reply TestResult.py]: Works for me everywhere I've tried it. :-)

Comment (andrewbonney) [reply TestResult.py]: Ok, I'll do that for now, but happy to re-visit if we come across a platform which handles things differently.

**Pull request merged**